wolfcrypt/src/random.c

```diff
@@ -198,58 +209,73 @@ static int Hash_df(DRBG* drbg, byte* out, word32 outSz, byte type,
                                                   const byte* inA, word32 inASz,
                                                   const byte* inB, word32 inBSz)
 {
+    int ret;
     byte ctr;
     int i;
     int len;
     word32 bits = (outSz * 8); /* reverse byte order */
     Sha256 sha;
-    byte digest[SHA256_DIGEST_SIZE];
+    DECLARE_VAR(digest, byte, SHA256_DIGEST_SIZE, drbg->heap);
```

Comment:
I don't like the fact that this is hiding a variable declaration, though obvious from the macro name, and potentially a malloc.
Yeah it's copy-pasting code if you remove the macro but I think it hides too much.
What does Todd think of this?